epoxy-processor/src/main/java/com/airbnb/epoxy/GeneratedModelWriter.kt

```diff
@@ -435,6 +484,74 @@ internal class GeneratedModelWriter(
         }
     }
 
+    private fun generateVisibilityMethods(modelInfo: GeneratedModelInfo): Iterable<MethodSpec> {
+        val methods = ArrayList<MethodSpec>()
+
+        val visibilityObjectParam = ParameterSpec.builder(modelInfo.modelType, "object", FINAL).build()
+
+        methods.add(buildVisibilityStateChangedMethod(visibilityObjectParam))
+
+        val onVisibilityStateChangedListenerType = ParameterizedTypeName.get(
+            getClassName(ON_VISIBILITY_STATE_MODEL_LISTENER_TYPE),
+            modelInfo.parameterizedGeneratedName,
+            modelInfo.modelType
+        )
+        val visibilityStateChangedListenerParam = ParameterSpec.builder(onVisibilityStateChangedListenerType, "listener").build()
+
+        val onVisibilityStateChanged = MethodSpec.methodBuilder("onVisibilityStateChanged")
+            .addJavadoc(
+                "Register a listener that will be called when this model visibility state "
+                        + "has changed.\n"
+                        + "<p>\n"
+                        + "The listener will contribute to this model's hashCode state per the {@link\n"
+                        + "com.airbnb.epoxy.EpoxyAttribute.Option#DoNotHash} rules.\n"
+                        + "<p>\n"
+                        + "You may clear the listener by setting a null value, or by calling "
```

Comment:
let's remove `You may clear the listener by setting a null value, or by calling {@link #reset()} */ `mutable models are not a pattern we encourage, and is the wrong advice for most people since so few use EpoxyAdapter